src/makeRobot.js

```diff
@@ -38,7 +38,91 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name,
+    wheels,
+    version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, `
+        + `chip version: ${this.version}, wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(step) {
+      if (step < 0) {
+        return this;
+      }
+
+      if (!step) {
+        this.coords.y += 1;
+
+        return this;
+      }
+      this.coords.y += step;
+
+      return this;
+    },
+
+    goBack(step) {
+      if (step < 0) {
+        return this;
+      }
+
+      if (!step) {
+        this.coords.y -= 1;
+
+        return this;
+      }
+      this.coords.y -= step;
+
+      return this;
+    },
+
+    goLeft(step) {
+      if (step < 0) {
+        return this;
+      }
+
+      if (!step) {
+        this.coords.x -= 1;
+
+        return this;
+      }
+      this.coords.x -= step;
+
+      return this;
+    },
+
+    goRight(step) {
+      if (step < 0) {
+        return this;
+      }
+
+      if (!step) {
+        this.coords.x += 1;
+
+        return this;
+      }
+      this.coords.x += step;
+
+      return this;
+    },
```

Comment:
```suggestion
    goRight(step = 1) {
      if (step > 0} this.coords.x += step;

      return this;
    },
```
please make it as simple as possible ⏫